### PR TITLE
Add example/template code for a first ImageJ plugin

### PR DESCRIPTION
**Pull request by haesleinhuepf**
Hi guys,

I would like to contribute this code for the very beginners of plugin development in ImageJ2. It may be seen as the IJ2 counterpart of this: https://github.com/imagej/minimal-ij1-plugin

New developers may download this tiny project folder and replace every appearance of "Enter your ... here" with their ... Then, they have their first own runnign IJ2 plugin in just 10 minutes.

These two files appear to me as the absolute minimum for a plugin in ImageJ2.

If you have any suggestions of essential stuff that's missing. I'm happy to add it.

Cheers,
Robert

### CODE REVIEW

File: my_first_plugin/pom.xml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0
+		http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+
+
+	<parent>
+		<groupId>net.imagej</groupId>
+		<artifactId>pom-imagej</artifactId>
```
Comment (imagejan) [inline my_first_plugin/pom.xml]: Should be `org.scijava / pom-scijava / 12.0.0` now, by default

File: my_first_plugin/src/main/java/net/enteryourdomainhere/enteryourprojectnamehere/EnterYourPluginNameHere.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import net.imglib2.RandomAccessibleInterval;
+import net.imglib2.img.Img;
+import net.imglib2.type.numeric.RealType;
+import org.scijava.command.Command;
+import org.scijava.plugin.Parameter;
+import org.scijava.plugin.Plugin;
+import org.scijava.ui.UIService;
+
+import net.imagej.Dataset;
+import net.imagej.ImageJ;
```
Comment (imagejan) [inline my_first_plugin/src/main/java/net/enteryourdomainhere/enteryourprojectnamehere/EnterYourPluginNameHere.java]: Sort the imports according to imagej project style?

File: my_first_plugin/src/main/java/net/enteryourdomainhere/enteryourprojectnamehere/EnterYourPluginNameHere.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        final Dataset dataset = ij.scifio().datasetIO().open(file.getPath());
+        System.out.println("hello2");
+
+        // Show the image
+        ij.ui().show(dataset);
+
+        // invoke the plugins run-function for testing...
+        ij.command().run(EnterYourPluginNameHere.class, true);
+    }
+}
```
Comment (imagejan) [inline my_first_plugin/src/main/java/net/enteryourdomainhere/enteryourprojectnamehere/EnterYourPluginNameHere.java]: Add line break at end of file.

File: my_first_plugin/pom.xml (hunk 1 above)
Comment (haesleinhuepf) [inline my_first_plugin/pom.xml]: I tried this already, but got error messages like:

[ERROR] /Users/rhaase/code/imagej-tutorials_my_first_plugin/my_first_plugin/src/main/java/net/enteryourdomainhere/enteryourprojectnamehere/EnterYourPluginNameHere.java:[6,22] package net.imagej.ops does not exist
[ERROR] /Users/rhaase/code/imagej-tutorials_my_first_plugin/my_first_plugin/src/main/java/net/enteryourdomainhere/enteryourprojectnamehere/EnterYourPluginNameHere.java:[7,19] package net.imglib2 does not exist
[ERROR] /Users/rhaase/code/imagej-tutorials_my_first_plugin/my_first_plugin/src/main/java/net/enteryourdomainhere/enteryourprojectnamehere/EnterYourPluginNameHere.java:[8,23] package net.imglib2.img does not exist
[ERROR] /Users/rhaase/code/imagej-tutorials_my_first_plugin/my_first_plugin/src/main/java/net/enteryourdomainhere/enteryourprojectnamehere/EnterYourPluginNameHere.java:[9,32] package net.imglib2.type.numeric does not exist
[ERROR] /Users/rhaase/code/imagej-tutorials_my_first_plugin/my_first_plugin/src/main/java/net/enteryourdomainhere/enteryourprojectnamehere/EnterYourPluginNameHere.java:[10,27] package org.scijava.command does not exist
[ERROR] /Users/rhaase/code/imagej-tutorials_my_first_plugin/my_first_plugin/src/main/java/net/enteryourdomainhere/enteryourprojectnamehere/EnterYourPluginNameHere.java:[11,26] package org.scijava.plugin does not exist
[ERROR] /Users/rhaase/code/imagej-tutorials_my_first_plugin/my_first_plugin/src/main/java/net/enteryourdomainhere/enteryourprojectnamehere/EnterYourPluginNameHere.java:[12,26] package org.scijava.plugin does not exist
[ERROR] /Users/rhaase/code/imagej-tutorials_my_first_plugin/my_first_plugin/src/main/java/net/enteryourdomainhere/enteryourprojectnamehere/EnterYourPluginNameHere.java:[13,22] package org.scijava.ui does not exist
[ERROR] /Users/rhaase/code/imagej-tutorials_my_first_plugin/my_first_plugin/src/main/java/net/enteryourdomainhere/enteryourprojectnamehere/EnterYourPluginNameHere.java:[15,18] package net.imagej does not exist
[ERROR] /Users/rhaase/code/imagej-tutorials_my_first_plugin/my_first_plugin/src/main/java/net/enteryourdomainhere/enteryourprojectnamehere/EnterYourPluginNameHere.java:[16,18] package net.imagej does not exist
[ERROR] /Users/rhaase/code/imagej-tutorials_my_first_plugin/my_first_plugin/src/main/java/net/enteryourdomainhere/enteryourprojectnamehere/EnterYourPluginNameHere.java:[25,72] cannot find symbol
  symbol: class Command

File: my_first_plugin/src/main/java/net/enteryourdomainhere/enteryourprojectnamehere/EnterYourPluginNameHere.java (hunk 2 above)
Comment (haesleinhuepf) [inline my_first_plugin/src/main/java/net/enteryourdomainhere/enteryourprojectnamehere/EnterYourPluginNameHere.java]: How should the order look like? I'm happy to change it, but I didn't find an explanation, e.g. here: http://imagej.net/Coding_style

File: my_first_plugin/src/main/java/net/enteryourdomainhere/enteryourprojectnamehere/EnterYourPluginNameHere.java (hunk 3 above)
Comment (haesleinhuepf) [inline my_first_plugin/src/main/java/net/enteryourdomainhere/enteryourprojectnamehere/EnterYourPluginNameHere.java]: Done.

File: my_first_plugin/src/main/java/net/enteryourdomainhere/enteryourprojectnamehere/EnterYourPluginNameHere.java (hunk 2 above)
Comment (rimadoma) [inline my_first_plugin/src/main/java/net/enteryourdomainhere/enteryourprojectnamehere/EnterYourPluginNameHere.java]: The order is: com;io;java;javax;net;org;

Comment (haesleinhuepf) [inline my_first_plugin/src/main/java/net/enteryourdomainhere/enteryourprojectnamehere/EnterYourPluginNameHere.java]: Just alphabetically?

File: my_first_plugin/pom.xml (hunk 1 above)
Comment (imagejan) [inline my_first_plugin/pom.xml]: 🤔  Hm, alright, then you'd have to declare all dependencies explicitly (i.e. `scijava-common`, `imglib2`, `imagej-ops`). But I guess plugin developers should anyways do that to be on the safe side, and from what I understood from [this lengthy discussion](http://forum.imagej.net/t/split-boms-from-parent-configuration/2563?u=imagejan), `pom-scijava` should be the new default parent for this kind of projects.

Comment (haesleinhuepf) [inline my_first_plugin/pom.xml]: When I make pom-scijava parent, it becomes an endless chain of complains by maven. Maybe there need to be some refinements first, before we can ask developers to use pom-scijava as parent...

File: my_first_plugin/src/main/java/net/enteryourdomainhere/enteryourprojectnamehere/EnterYourPluginNameHere.java (hunk 2 above)
Comment (haesleinhuepf) [inline my_first_plugin/src/main/java/net/enteryourdomainhere/enteryourprojectnamehere/EnterYourPluginNameHere.java]: Didn't work in my Intellij ;-) but I guess its alright now.

**Comment by ctrueden** [comment]
Thanks @haesleinhuepf for contributing this, and sorry for the neglect.

I am going to close this, because we now have:

    https://github.com/imagej/example-imagej-command

Which hopefully fulfills your goals in creating this.

This tutorials repository is being repositioned as a collection of tutorial notebooks, rather than Java code.

Please take a look at the `example-imagej-command` and let us know what you think!

File: my_first_plugin/pom.xml (hunk 1 above)
Comment (ctrueden) [inline my_first_plugin/pom.xml]: @haesleinhuepf I would like to hear about what sort of "endless chain of complaints" you mean. The `pom-scijava` is definitely the sole intended parent for everyone now. It is more stringent than before, but there are certainly reasons for that. Hopefully starting from the `example-imagej-command` project, editing its POM as indicated, alleviates your problems?

**Pull request closed without merging**